src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/RazorConnectHandler.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.LanguageServer.EndpointContracts;
+using Microsoft.CodeAnalysis.Razor.Protocol;
+using Microsoft.CommonLanguageServerProtocol.Framework;
+using Microsoft.VisualStudio.Threading;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem;
+
+[RazorLanguageServerEndpoint(CustomMessageNames.RazorConnectEndpointName)]
+internal class RazorConnectHandler(NamedPipeBasedRazorProjectInfoDriver infoDriver) : IRazorNotificationHandler<RazorConnectParams>
+{
+    public bool MutatesSolutionState => false;
+
+    public Task HandleNotificationAsync(RazorConnectParams request, RazorRequestContext requestContext, CancellationToken cancellationToken)
+    {
+        infoDriver.ConnectAsync(request.PipeName, cancellationToken).Forget();
```

Comment:
Why are we forgetting this `Task` here? Seems very odd at a glance to be calling an `async` method, forgetting the `Task` from within an async method. Can we comment why this is happening?

Reply:
When I was testing I was hitting an issue where this task would not complete. I can change this to return the task instead of forgetting it but opted for this approach because it was easier to diagnose when debugging. Basically bringing all LSP to a halt vs having this one system broken

Reply:
Worst case add a comment explaining this might be a hack?

Reply:
I just removed this and made sure the connect async does not return the long running task